File: ch8/js/e-resume.js
import { MemorySaver } from "@langchain/langgraph";
import { StateGraph } from "@langchain/langgraph";
import { Annotation } from "@langchain/langgraph";
import * as dotenv from "dotenv/config";
dotenv;

const GraphState = Annotation.Root({
  messages: Annotation(),
});

// Assuming graph is already created and configured
const graph = new StateGraph(GraphState).compile({
  checkpointer: new MemorySaver(),
});

const config = { configurable: { thread_id: "1" } };

const output = await graph.stream(null, {
  ...config,
  interruptBefore: ["tools"],
});

for await (const chunk of output) {
  console.log(chunk); // do something with the output
}

File: ch8/py/e-resume.py
from langgraph.graph import StateGraph
from langgraph.checkpoint.memory import MemorySaver


async def main():
    # Create a simple graph
    builder = StateGraph()
    # Add nodes and edges as needed
    graph = builder.compile(checkpointer=MemorySaver())

    config = {"configurable": {"thread_id": "1"}}

    output = graph.astream(None, config, interrupt_before=["tools"])

    async for c in output:
        print(c)  # do something with the output


if __name__ == "__main__":
    import asyncio

    asyncio.run(main())
